experimental/oak_baremetal_loader/src/crosvm.rs

```diff
@@ -133,9 +85,25 @@ impl Vmm for Crosvm {
         self.wait().await
     }
 
-    fn create_comms_channel(&self) -> Result<Box<dyn ReadWrite>> {
-        // Create the channel in a disconnected state as it can only be connected later, after the
-        // VM has booted up.
-        Ok(Box::new(VsockStreamChannel { stream: None }))
+    async fn create_comms_channel(&self) -> Result<Box<dyn ReadWrite>> {
+        // The vsock channel can only be created after the VM is booted. Hence
+        // we try a few times to connect, waiting for the VM to be ready.
+        // If no connetion is established after a while, we timeout.
+        let task = tokio::spawn(async {
+            let stream: Box<dyn ReadWrite> = loop {
+                match VsockStream::connect_with_cid_port(VSOCK_GUEST_CID, VSOCK_GUEST_PORT) {
+                    Ok(stream) => {
+                        break Box::new(stream);
+                    }
+                    Err(_error) => {
+                        sleep(Duration::from_millis(100)).await;
+                        continue;
+                    }
+                }
+            };
+            stream
+        })
+        .map_err(anyhow::Error::msg);
+        timeout(Duration::from_millis(1000), task).await?
```

Comment:
For simultaneous Read/Write in the future, the plan is to update this function to return seperate Read and Write halves (both our virtio vsock & unix stream let us clone the handle). To do that though, the channel must be connected.